File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.py
def bfs(s, e):
    q=[]
    v=[0]*200001

    q.append(s)
    v[s]=1

    while q:
        # ch,ci,cj = q.pop(0) # deque 썼으니까 popleft로 바꿔줌
        c = q.pop(0)
        if c==e: # 목적지 도착
            return v[e]-1
        # 3방향(-1, +1, *2), 범위내(0<= <=200000), 미방문
        for n in (c-1, c+1, c*2):
            if 0<=n<=200000 and v[n]==0:
                q.append(n)
                v[n]=v[c]+1

N, K = map(int, input(). split()) # 수빈이와 동생의 위치 
answer=bfs(N, K)
print(answer)

File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split(" ");

const n = parseInt(input[0]);
const k = parseInt(input[1]);

function bfs(s, e) {
  let q = [s];
  let v = Array.from({ length: 200001 }, () => 0);
  v[s] = 1;

  while (q.length) {
    let c = q.shift();

    if (c === e) return v[e] - 1;

    for (let n of [c - 1, c + 1, c * 2]) {
      if (0 <= n && n <= 200000 && v[n] === 0) {
        q.push(n);
        v[n] = v[c] + 1;
      }
    }
  }
}

let answer = bfs(n, k);
console.log(answer);
